components/hook/hash-change.js

```diff
@@ -0,0 +1,21 @@
+import { useEffect } from 'react'
+
+const getHash = () => window.location.hash
+
+let currentHash
```

Comment:
Although this is working like a charm I'm not sure if a more logical workflow could be using `React.Context` API.